### ISSUE TITLE
`fallbackToSettings` is not used in "OneSignal#promptPermission" method call

### ISSUE THREAD

**Issue by om-ha** [OneSignal/OneSignal-Flutter-SDK#234]
# Problem
**Description:**
As the title suggests, _OneSignal-Flutter-SDK_ does exposes `fallbackToSettings` but does not actually pass it to iOS SDK.

**Environment**
pub -- onesignal_flutter: ^2.3.4
Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.14.5 18F132, locale en-US)
iOS 13.3
Xcode - develop for iOS and macOS (Xcode 11.3)

**Steps to Reproduce Issue:**

1. Attempt to call `OneSignal.shared.promptUserForPushNotificationPermission(fallbackToSettings: true);`
1. Native notification permission prompt appears
1. Cancel/Deny permission from said prompt
1. Re-call method in #1, passing `fallbackToSettings` as `true`
1. Prompt/Settings do not appear

# Proposed Fix
**1.** [OneSignal-**Flutter**-SDK] `fallbackToSettings` flag is exposed in flutter code via [promptUserForPushNotificationPermission](https://github.com/OneSignal/OneSignal-Flutter-SDK/blob/master/lib/onesignal_flutter.dart#L145
). This shouldn't require any modifications.

**2.** [OneSignal-**Flutter**-SDK] `fallbackToSettings` is **NOT** passed to this method nor exposed in this native iOS code, this should be done [here (1)](https://github.com/OneSignal/OneSignal-Flutter-SDK/blob/master/ios/Classes/OneSignalPlugin.m#L125) and [here (2)](https://github.com/OneSignal/OneSignal-Flutter-SDK/blob/master/ios/Classes/OneSignalPlugin.m#L208). This is where the problem lies, this is where the modifications should be made based on step **3.** below.

**3.** [OneSignal-**iOS**-SDK] The native iOS method [`promptForPushNotificationsWithUserResponse`](https://github.com/OneSignal/OneSignal-iOS-SDK/blob/master/iOS_SDK/OneSignalSDK/Source/OneSignal.m#L917) exposes/takes `fallbackToSettings` flag. Also see the [test here](https://github.com/OneSignal/OneSignal-iOS-SDK/blob/master/iOS_SDK/OneSignalSDK/UnitTests/UnitTests.m#L2197) for reference.

**Comment by rgomezp** [comment]
Thanks for your suggestion. We will consider it for a future release

**Comment by MichaelTamm** [comment]
Please fix this. I just wasted some hours of work time because of this issue.

**Comment by MichaelTamm** [comment]
And I also don't think, this issue is labeled correctly: This is a **bug**, because when calling
```
OneSignal.shared.promptUserForPushNotificationPermission(fallbackToSettings: true);
```
The `fallbackToSettings: true` does **not** work.

By the way, I'm using `onesignal_flutter 3.0.0-beta4`

**Comment by AlirezaDaryani** [comment]
I used `onesignal_flutter: 3.0.0` and `fallbackToSettings` NOT work.

**Comment by AlirezaDaryani** [comment]
Any update?!
We need this dear team.

**Comment by nan-li** [comment]
Hi all,
Sorry for the delay and thank you for waiting.
A PR was submitted in https://github.com/OneSignal/OneSignal-Flutter-SDK/pull/523 to fix this, and it has been released in [onesignal_flutter `3.3.0`](https://github.com/OneSignal/OneSignal-Flutter-SDK/releases/tag/3.3.0).

**Comment by om-ha** [comment]
At last! Thank you.

It's been over 2 years 🙈 😅
